### PR TITLE
fix: adding clear button for datepicker

### LINKED ISSUE #5835: DatePicker: Add optional "Clear" button
Add CLEAR button where user quickly clear the selected value from the date-picker control.

### PR DESCRIPTION
**Pull request by UdayAppam**
closes #5835

DatePicker: Add optional "Clear" button
# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [ ] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [ ] built and tested the changes locally.
 - [ ] added/updated tests.
 - [ ] added/updated API documentation.
 - [ ] added/updated demos.

### CODE REVIEW

**Comment by UdayAppam** [comment]
@daniloff200 
Could you please help me to fix the clear button unit test case issue

**Comment by daniloff200** [comment]
Hey @KernelLora

Well, at current moment fixing all the jobs on Travis can be complicated, because, Angular 10 is already released and some jobs (with Next and Latest in their names) are using it, and, ngx-bootstrap right now does not support NG10 (we're in progress of it)

But, I see, that you also have a Lint errors, so, first of all, you should fix them.
Also, you added a new functionality to the datepicker, so, it will be very important to add a demo case for that, with a description, how to use it, and how it works, for the documentation
And after that we can proceed with fixing of some broken tests

File: src/datepicker/bs-datepicker.spec.ts [hunk 1]
```diff
@@ -165,4 +165,24 @@ describe('datepicker:', () => {
         expect(view.date.getDay()).toEqual((new Date().getDay()), 'should update to equal today');
       }).unsubscribe();
   });
+
+
+  fit('should clear date', () => {
```
Comment (daniloff200) [inline src/datepicker/bs-datepicker.spec.ts]: please, remove **f**

Comment (UdayAppam) [reply src/datepicker/bs-datepicker.spec.ts]: Sorry @daniloff200 
removed the **f**
Could you please help me to fix this test case for the clear date

**Comment by UdayAppam** [comment]
@daniloff200 
Could you please review this and if needed any changes please guide me.
May I know when can we expect  Angular 10 support ?
Accordingly my company projects upgrades will plan.

**Comment by UdayAppam** [comment]
Need help to fix the unit test.
Could anyone please check

**Comment by UdayAppam** [comment]
@daniloff200 
Seems to be all checks passed.
Could you please review once

**Review by daniloff200 — APPROVED** [review]

**Pull request merged**